lib/bundle.js

```diff
@@ -20,158 +20,253 @@ var Builder = require('systemjs-builder');
 var systemBuilder = new Builder();
 var fs = require('fs');
 var asp = require('rsvp').denodeify;
+var glob = require('glob');
 
 exports.depCache = function(moduleName) {
   return config.load()
-  .then(function() {
-    moduleName = moduleName || config.loader.main;
-  })
-  .then(function() {
-    ui.log('info', 'Injecting the traced dependency tree for `' + moduleName + '`...');
-
-    // trace the starting module
-    var cfg = config.loader.getConfig();
-    cfg.baseURL = path.relative(process.cwd(), config.pjson.baseURL);
-
-    return systemBuilder.trace(moduleName, cfg);
-  })
-  .then(function(output) {
-    var traceTree = output.tree;
-    moduleName = output.moduleName;
-    var depCache = config.loader.depCache = config.loader.depCache || {};
-    for (var d in traceTree) {
-      var deps = traceTree[d].deps.map(function(dep) {
-        return traceTree[d].depMap[dep];
-      });
+    .then(function() {
+      moduleName = moduleName || config.loader.main;
+    })
+    .then(function() {
+      ui.log('info', 'Injecting the traced dependency tree for `' + moduleName + '`...');
```

Comment:
I hate to be pedantic about style, but can we keep the `.then` flush with the initial statement? I know it's not the common style, but the rest of the codebase is done this way.